### PR TITLE
Change GKE to depend on the latest version of weaver.

### PR DESCRIPTION
**Pull request by spetrovic77**
Multiple changes, grouped as follows:
  * Update the babysitter given the new pipe protocol.
  * Update the babysitter <-> manager protocol given the changes to runtime.proto.
  * Rename "group" -> "ReplicaSet", now that the base weaver no longer references colocation groups.
  * weaver.Logger -> slog.Logger changes.
  * Fix the dashboard.
  * Various drive-by fixes.

### CODE REVIEW

File: internal/nanny/manager/manager_test.go [hunk 1]
```diff
@@ -235,15 +232,15 @@ func TestGetRoutingInfo(t *testing.T) {
...
 		}
-		sort.Strings(reply.Replicas)
+		sort.Strings(reply.Routing.Replicas)
 		return &reply
 	}
 
-	routingInfo := resolve("")
-	if len(routingInfo.Replicas) != 0 {
-		t.Fatalf("resolve(empty) = %v; want []", routingInfo.Replicas)
+	routingInfo := resolve("", "nonexistant")
```
Comment (mwhittaker) [inline internal/nanny/manager/manager_test.go:239]: "nonexistent"

File: internal/nanny/manager/manager_test.go [hunk 2]
```diff
@@ -258,17 +255,30 @@ func TestGetRoutingInfo(t *testing.T) {
...
-	if diff := cmp.Diff(testData[:2], routingInfo.Replicas); diff != "" {
+	routingInfo = resolve("", "unsharded")
+	if diff := cmp.Diff(testData[:2], routingInfo.Routing.Replicas); diff != "" {
 		t.Fatalf("unexpected resolver result (-want,+got):\n%s", diff)
 	}
 	if routingInfo.Version == "" {
 		t.Fatal("no version returned by resolver")
 	}
-	assignment, _ := assigner.FromProto(routingInfo.Assignments[0])
+	if routingInfo.Routing.Assignment != nil {
```
Comment (mwhittaker) [inline internal/nanny/manager/manager_test.go:268]: In https://github.com/ServiceWeaver/weaver/pull/222, I add a `routing.FormatAssignment` function which might be useful here. It should pretty print assignments in a more readable way than `prototext.Format`. Just a note for the future.

File: internal/nanny/assigner/assigner.go [hunk 3]
```diff
@@ -181,62 +176,54 @@ func (a *Assigner) registerApp(ctx context.Context, app string) error {
...
 	}
 	return nil
 }
 
-func (a *Assigner) RegisterComponentToStart(ctx context.Context, req *protos.ComponentToStart) error {
-	id, err := uuid.Parse(req.DeploymentId)
+// RegisterActiveComponent registers the given component has been activated. It
+// returns the name of the Kubernetes ReplicaSet that should host the component.
+func (a *Assigner) RegisterActiveComponent(ctx context.Context, req *nanny.ActivateComponentRequest) error {
+	id, err := uuid.Parse(req.Config.Deployment.Id)
```
Comment (mwhittaker) [inline internal/nanny/assigner/assigner.go:200]: In a future change, we can avoid parsing the uuid and update the `store.` methods?

File: internal/local/starter.go [hunk 4]
```diff
@@ -61,16 +65,18 @@ func (s *Starter) Start(ctx context.Context, cfg *config.GKEConfig, group *proto
...
-	babysitters := make([]*babysitter.Babysitter, defaultReplication)
+	babysitters := make([]bsitter, defaultReplication)
 	for i := 0; i < defaultReplication; i++ {
-		b, err := createBabysitter(ctx, cfg, group, LogDir)
+		ctx, cancel := context.WithCancel(ctx)
+		b, err := createBabysitter(ctx, cfg, replicaSet, LogDir)
 		if err != nil {
 			// TODO(mwhittaker): Should we stop the previously started
 			// babysitters?
+			cancel()
```
Comment (mwhittaker) [inline internal/local/starter.go:75]: Nice.

**Review by mwhittaker — APPROVED** [review]
Amazing job Srdjan! Absolutely heroic change :superhero:

File: internal/babysitter/client.go [hunk 5]
```diff
@@ -47,10 +47,10 @@ func (h *HttpClient) RunProfiling(ctx context.Context, req *protos.RunProfiling)
 }
 
 // CheckHealth implements the clients.BabysitterClient interface.
-func (h *HttpClient) CheckHealth(ctx context.Context, req *clients.HealthCheck) (*protos.HealthReport, error) {
```
[the remark this answers is not in the document]
Comment (spetrovic77) [reply internal/babysitter/client.go:50]: They aren't needed any more. We now do it through the context, on the client-side (i.e., the assigner).

File: internal/nanny/manager/manager_test.go (hunk 2 above)
Comment (spetrovic77) [reply internal/nanny/manager/manager_test.go:268]: ACK, thanks.

File: internal/nanny/controller/controller.go [hunk 6]
```diff
@@ -535,78 +531,20 @@ func (c *controller) runProfiling(ctx context.Context, v *AppVersionState, req *
...
-	wait.Add(len(states))
-	for _, state := range states {
-		state := state
-		go func() {
-			defer wait.Done()
-			p, err := state.d.RunProfiling(ctx, req)
-			if err != nil {
-				state.errs = append(state.errs, err.Error())
-				return
+	groups := make([][]func() ([]byte, error), 0, len(addrs))
```
[the remark this answers is not in the document]
Comment (spetrovic77) [reply internal/nanny/controller/controller.go:534]: Thanks!

File: internal/nanny/nanny.proto [hunk 7]
```diff
@@ -93,12 +86,83 @@ message ApplicationDeleteRequest {
...
+message GetComponentsReply {
+  string version = 1;              // Version of the current component set.
+  repeated string components = 2;  // A list of components to run.
+}
+
+// GetRoutingInfo is a request to the manager to retrieve routing information
+// for a given component.
+message GetRoutingRequest {
+  string component = 1;         // Component that needs to be activated.
+  bool routed = 2;              // Is the component routed?
```
[the remark this answers is not in the document]
Comment (spetrovic77) [reply internal/nanny/nanny.proto:125]: Good find. Deleted.

File: internal/nanny/assigner/assigner.go (hunk 3 above)
Comment (spetrovic77) [reply internal/nanny/assigner/assigner.go:200]: Thanks. Added a TODO. WIl do so in subsequent change.

File: internal/babysitter/client.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (mwhittaker) [reply internal/babysitter/client.go:50]: Ah got it.

**Pull request merged**